applications/RANSApplication/custom_utilities/rans_variable_utilities.cpp

```diff
@@ -105,33 +84,11 @@ double GetMinimumScalarValue(
     const auto& r_communicator = rModelPart.GetCommunicator();
     const auto& r_nodes = r_communicator.LocalMesh().Nodes();
 
-    class MinReducer{
-        public:
-            typedef double value_type;
-            double min_value = std::numeric_limits<double>::max();
-
-            value_type GetValue()
-            {
-                return min_value;
-            }
-
-            void LocalReduce(const double min_value){
-                this->min_value = std::min(this->min_value, min_value);
-            }
-            void ThreadSafeReduce(MinReducer& rOther){
-                #pragma omp critical
-                {
-                    this->min_value = std::min(this->min_value, rOther.min_value);
-                }
-            }
-    };
-
     const int number_of_nodes = r_nodes.size();
     const double min_value =
-        IndexPartition<int>(number_of_nodes).for_each<MinReducer>(
-            [&](const int i) -> double {
-                return (r_nodes.begin() + i)->FastGetSolutionStepValue(rVariable);
-            });
+        IndexPartition<int>(number_of_nodes).for_each<MinReduction<double>>([&](const int i) -> double {
+            return (r_nodes.begin() + i)->FastGetSolutionStepValue(rVariable);
+        });
```

Comment:
I think you can use the `block_for_each` in here.

Reply:
again the problem of `const ModelPart`